### ISSUE TITLE
bcftools csq gives 2 mutations for same position in case of inflame_insertion and missense

### ISSUE THREAD

**Issue by MWendorff** [samtools/bcftools#1536]
Hi again,

when calling csq on some mutation combinations I receive two different consequences for the same amino acid/transcript/strain/patient. In the following example the consequence 216SC* would be ecpected but I receive the following two modifications.

2       31387815        2_31387815_A_AGC        A       AGC     621     PASS    AF=2.5e-05;AQ=621;ExcessHet=3.0103;QD=14.79;AC=1;AN=2;BCSQ=frameshift|XDH|ENST00000379416|protein_coding|-|**216L**LRLKDTPRKQLRFEGERVTWIQASTLKELLDLKAQHPDAKLVVGNTEIGIEMKFKNMLFPMIVCPAWIPELNSVEHGPDGISFGAACPLSIVEKTLVDAVAKLPAQKTEVFRGVLEQLRWFAGKQVKSVASVGGNIITASPISDLNPVFMASGAKLTLVSRGTRRTVQMDHTFFPGYRKTLLSPEEILLSIEIPYSREGEYFSAFKQASRREDDIAKVTSGMRVLFKPGTTEVQELALCYGGMANRTISALKTTQRQLSKLWKEELLQDVCAGLAEELHLPPDAPGGMVDFRCTLTLSFFFKFYLTVLQKLGQENLEDKCGKLDPTFASATLLFQKDPPADVQLFQEVPKGQSEEDMVGRPLPHLAADMQASGEAVYCDDIPRYENELSLRLVTSTRAHAKIKSIDTSEAKKVPGFVCFISADDVPGSNITGICNDETVFAKDKVTCVGHIIGAVVADTPEHTQRAAQGVKITYEELPAIITIEDAIKNNSFYGPELKIEKGDLKKGFSEADNVVSGEIYIGGQEHFYLETHCTIAVPKGEAGEMELFVSTQNTMKTQSFVAKMLGVPANRIVVRVKRMGGGFGGKETRSTVVSTAVALAAYKTGRPVRCMLDRDEDMLITGGRHPFLARYKVGFMKTGTVVALEVDHFSNVGNTQDLSQSIMERALFHMDNCYKIPNIRGTGRLCKTNLPSNTAFRGFGGPQGMLIAECWMSEVAVTCGMPAEEVRRKNLYKEGDLTHFNQKLEGFTLPRCWEECLASSQYHARKSEVDKFNKENCWKKRGLCIIPTKFGISFTVPFLNQAGALLHVYTDGSVLLTHGGTEMGQGLHTKMVQVASRALKIPTSKIYISETSTNTVPNTSPTAASVSADLNGQAVYAACQTILKRLEPYKKKNPSGSWEDWVTAAYMDTVSLSATGFYRTPNLGYSFETNSGNPFHYFSYGVACSEVEIDCLTGDHKNLRTDIVMDVGSSLNPAIDIGQVEGAFVQGLGLFTLEELHYSPEGSLHTRGPSTYKIPAFGSIPIEFRVSLLRDCPNKKAIYASKAVGEPPLFLAASIFFAIKDAIRAARAQHTGNNVKELFRLDSPATPEKIRNACVDKFTTLCVTGVPENCKPWSVRV*>**216C**C*|31387815A>AGC      GT:AB:AD:DP:GQ:PL:RNC:SBPV:BCSQ 0/1:0.666667:28,14:42:99:621,0,1358:..:0.1341:2

2       31387816        2_31387816_A_T  A       T       517     PASS    AF=2.5e-05;AQ=517;ExcessHet=3.0103;QD=12.02;AC=1;AN=2;BCSQ=missense|XDH|ENST00000379416|protein_coding|-|**216L>216M**|31387816A>T  GT:AB:AD:DP:GQ:PL:RNC:SBPV:BCSQ 0/1:0.636364:28,15:44:99:517,0,947:..:0.2254:2

Description of the case:
- frameshift_mutation with an insertion of 2 bases (A->AGC)
- directly followed by a missense mutation (A->T)
- should combining to CAG|CAA|CTC -> CA|GCA|**GCT**|CTC (NOTE: manually I would most probably define it as one mutation A->GCT, but the used pipeline results in two mutations and the called consequence is wrong.)
- would result to the amino acid sequence 215ESC*

We also figured out another case: The architecture of the mutations is the same as in this case:
- frameshift_mutation with an insertion of 2 bases (G->GTT)
- directly followed by a missense mutation (G->C)
- combining to CCA|GGG|**G**GA|A -> C|CAG|G**TT|C**GA|A

I used version:
bcftools 1.13-3-g89a566b
Using htslib 1.13-3-gd16bed5

I attach a small bcf file containing the two mutations (without consequences).
It would be great if you could solve this issue.

Best
Mareike
[subset_csqProblem2.bcf.gz](https://github.com/samtools/bcftools/files/6894131/subset_csqProblem2.bcf.gz)

**Comment by pd3** [comment]
What the `csq` command looks like? Can you make the relevant part of the gff file available as well please?

**Comment by MWendorff** [comment]
Hi Petr,

I used:

bcftools csq -p a --ncsq 128 -f Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -O b -o subset_csqProblem2_csq.bcf.gz -g Homo_sapiens.GRCh38.100.gff3.gz subset_csqProblem2.bcf.gz

**Comment by MWendorff** [comment]
The gff file is from http://ftp.ensembl.org/pub/release-100/gff3/homo_sapiens/

**Comment by pd3** [comment]
Thank you. I believe this is fixed by e0e3484. Please try it out and let me know if you spot more problems. Thank you for reporting the problem.

**Comment by MWendorff** [comment]
Hi Petr,
I am confused, as the current commit does not change anything in the output as far as I see. But the bcftools version is now "bcftools 1.13-9-ge0e3484" even if installed from commit e0e3484.

Best
Mareike

**Comment by pd3** [comment]
The two variants are now correctly recognized as compound, the first record is now:
```
2   31387815    2_31387815_A_AGC    A   AGC 621 PASS    AF=2.5e-05;AQ=621;ExcessHet=3.0103;QD=14.79;AC=1;AN=2;BCSQ=@31387816
```

Is it possible that you are running the previous version instead? If you are running the latest github checkout, he header of the output file show now contain:
```
##bcftools/csqVersion=1.13-10-gc75a420
```

**Comment by MWendorff** [comment]
Hi Petr,
thanks a lot. I still do not know why the old version is used on the old version. This case was solved now.
Unfortunately I found another very similar case which was not solved with this bug, where an inframe insertion and missense mutation affect the same codon.
Find the corresponding case attached.
[subset_csqProblem2b.bcf.gz](https://github.com/samtools/bcftools/files/6954527/subset_csqProblem2b.bcf.gz)
